packages/@aws-cdk/aws-amplify/lib/app.ts

```diff
@@ -375,30 +375,95 @@ export enum RedirectStatus {
 }
 
 /**
- * Custom rewrite/redirect rule for an Amplify App.
+ * Options for a custom rewrite/redirect rule for an Amplify App.
  */
-export interface CustomRule {
+export interface CustomRuleOptions {
   /**
    * The source pattern for a URL rewrite or redirect rule.
+   *
+   * @see https://docs.aws.amazon.com/amplify/latest/userguide/redirects.html
    */
   readonly source: string;
 
   /**
    * The target pattern for a URL rewrite or redirect rule.
+   *
+   * @see https://docs.aws.amazon.com/amplify/latest/userguide/redirects.html
    */
   readonly target: string
 
   /**
    * The status code for a URL rewrite or redirect rule.
    *
+   * @see https://docs.aws.amazon.com/amplify/latest/userguide/redirects.html
+   *
    * @default PERMANENT_REDIRECT
    */
   readonly status?: RedirectStatus;
 
   /**
    * The condition for a URL rewrite or redirect rule, e.g. country code.
    *
+   * @see https://docs.aws.amazon.com/amplify/latest/userguide/redirects.html
+   *
    * @default - no condition
    */
   readonly condition?: string;
 }
+
+/**
+ * Custom rewrite/redirect rule for an Amplify App.
+ *
+ * @see https://docs.aws.amazon.com/amplify/latest/userguide/redirects.html
+ */
+export class CustomRule {
+  /**
+   * Sets up a 200 rewrite for all files to `index.html` except for the
+   * following file extensions: css, gif, ico, jpg, js, png, txt, svg,
+   * woff, ttf, map, json, webmanifest.
+   */
+  public static readonly SINGLE_PAGE_APPLICATION_REDIRECT = new CustomRule({
```

Comment:
I've always found these lists of extensions to be problematic. In my experience a custom redirect rule for all paths that don't contain an explicit file extension is much easier to deal with and wouldn't require us to maintain this ongoing list.

@jogold what do you think?

Reply:
I think you're right... the regex comes from the docs at https://docs.aws.amazon.com/amplify/latest/userguide/redirects.html and I already had to add the `webmanifest` extension.

Reply:
Hopefully its relatively straightforward. I see this being relied on a lot. I'm actually surprised amplify doesn't have some solution built in for this behavior.